lib/pgBackRest/Config/Config.pm

```diff
@@ -2691,8 +2691,103 @@ sub optionValidate
             confess &log(ERROR, "option '${strOption}' not valid for command '${strCommand}'", ERROR_OPTION_COMMAND);
         }
     }
+
+    # If a config file was loaded, and the command is not a time-sensitive command then determine if all options are valid in the
+    # config file
+    if (defined($oConfig) && !($strCommand eq CMD_ARCHIVE_PUSH || $strCommand eq CMD_ARCHIVE_GET))
+    {
+        configFileValidate($oConfig);
+    }
+}
+
+####################################################################################################################################
+# configFileValidate
+#
+# Determine if the configuration file contains any invalid options or placements
+####################################################################################################################################
+sub configFileValidate
+{
+    my $oConfig = shift;
+
+    my $bFileValid = true;
+
+    foreach my $strSectionKey (keys(%$oConfig))
+    {
+        my ($strSection, $strCommand) = ($strSectionKey =~ m/(\w+):*(\w*-*\w*)/);
```

Comment:
https://msdn.microsoft.com/en-us/windows/hardware/drivers/install/general-syntax-rules-for-inf-files#section-names indicates the format of section names and allows for string tokens (e.g. %sectionname%) and such. Should we be validating the section name against the Windows INI file formatting conventions?